### PR TITLE
arch_timer: adjust timer/arch_timer to support tick

### PR DESCRIPTION
**Pull request by zyfeier**
## Summary

Enable CONFIG_SCHED_TICKLESS_TICK_ARGUMENT in tickless mode to improve the performance.

## Impact

NA

## Testing

bl602

### CODE REVIEW

File: include/nuttx/timers/timer.h [hunk 1]
```diff
@@ -200,6 +247,98 @@ extern "C"
...
+static inline
+int timer_tick_maxtimeout(FAR struct timer_lowerhalf_s *lower,
+                          FAR uint32_t *maxtimeout)
+{
+  uint32_t maxtimeout_us;
+  int ret;
+
+  DEBUGASSERT(lower->ops->maxtimeout);
+
+  ret = lower->ops->maxtimeout(lower, &maxtimeout_us);
```
Comment (xiaoxiang781216) [inline include/nuttx/timers/timer.h:333]: let's use maxtimeout directly

File: include/nuttx/timers/timer.h [hunk 2]
```diff
@@ -200,6 +247,98 @@ extern "C"
...
+static inline
+int timer_tick_getstatus(FAR struct timer_lowerhalf_s *lower,
+                          FAR struct timer_status_s *status)
+{
+  struct timer_status_s us_status;
+  int ret;
+
+  DEBUGASSERT(lower->ops->getstatus);
+
+  ret = lower->ops->getstatus(lower, &us_status);
```
Comment (xiaoxiang781216) [inline include/nuttx/timers/timer.h:305]: let's use status directly

File: include/nuttx/timers/timer.h [hunk 3]
```diff
@@ -200,6 +247,98 @@ extern "C"
...
+static inline
+int timer_maxtimeout(FAR struct timer_lowerhalf_s *lower,
+                     FAR uint32_t *maxtimeout)
+{
+  uint32_t maxtimeout_tick;
+  int ret;
+
+  DEBUGASSERT(lower->ops->tick_maxtimeout);
+
+  ret = lower->ops->tick_maxtimeout(lower, &maxtimeout_tick);
```
Comment (xiaoxiang781216) [inline include/nuttx/timers/timer.h:287]: let's use maxtimeout directly

File: include/nuttx/timers/timer.h [hunk 4]
```diff
@@ -200,6 +247,98 @@ extern "C"
...
+static inline
+int timer_getstatus(FAR struct timer_lowerhalf_s *lower,
+                     FAR struct timer_status_s *status)
+{
+  struct timer_status_s tick_status;
+  int ret;
+
+  DEBUGASSERT(lower->ops->tick_getstatus);
+
+  ret = lower->ops->tick_getstatus(lower, &tick_status);
```
Comment (xiaoxiang781216) [inline include/nuttx/timers/timer.h:259]: let's use status directly

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by raiden00pl** [comment]
@zyfeier @xiaoxiang781216 it looks like this PR breaks timer implementations that not define some ops (like getstatus() in stm32). Here's a problem on the mailing list: https://lists.apache.org/thread/sys37yf63rq501fd1v8y3zyh6vk10v1d

**Comment by xiaoxiang781216** [comment]
> @zyfeier @xiaoxiang781216 it looks like this PR breaks timer implementations that not define some ops (like getstatus() in stm32). Here's a problem on the mailing list: https://lists.apache.org/thread/sys37yf63rq501fd1v8y3zyh6vk10v1d

but getstatus is required for arch_timer even before this change.

**Comment by raiden00pl** [comment]
@xiaoxiang781216 this is the case when we don't use ARCH_TIMER. Only CONFIG_TIMER=y and for that case getstatus() is optional which is not handled by:
https://github.com/apache/nuttx/blob/afb368afd2e9e897176605f0a71357b1e9f56cda/include/nuttx/timers/timer.h#L103-L104

**Comment by xiaoxiang781216** [comment]
so, the assert happen when handling TCIOC_GETSTATUS in timer_ioctl?

**Comment by xiaoxiang781216** [comment]
Fix here: https://github.com/apache/nuttx/pull/12484

**Pull request merged**